### PR TITLE
Akka typed first documentation user experience

### PR DESCRIPTION
**Pull request by seglo**
References #984

### CODE REVIEW

File: tests/src/test/scala/docs/scaladsl/ConsumerExample.scala [hunk 1]
```diff
@@ -375,6 +377,67 @@ class ConsumerExample extends DocsSpecBase with TestcontainersKafkaLike {
...
+      case (context, TopicPartitionsRevoked(subscription, topicPartitions)) =>
+        context.log.info("Revoked: {}", topicPartitions)
+    //#withTypedRebalanceListenerActor
+        revokedPromise.success(Done)
+    //#withTypedRebalanceListenerActor
+        Behaviors.same
+    }
+    //#withTypedRebalanceListenerActor
+
+    val guardian: Behavior[Nothing] = Behaviors.setup[Nothing] { context =>
```
Comment (seglo) [inline tests/src/test/scala/docs/scaladsl/ConsumerExample.scala:407]: Tabbing is weird to line up columns in docs. Is there a better way?

**Comment by seglo** [comment]
@ennru I haven't gone through all the docs yet, but is this what you had in mind for updating the docs for a Akka 2.6 (typed) first experience?

File: build.sbt [hunk 2]
```diff
@@ -204,6 +204,7 @@ lazy val core = project
     name := "akka-stream-kafka",
     AutomaticModuleName.settings("akka.stream.alpakka.kafka"),
     libraryDependencies ++= Seq(
+        "com.typesafe.akka" %% "akka-actor-typed" % akkaVersion,
```
Comment (ennru) [inline build.sbt:207]: Hm, I'm not sure we want to add the dependency.
My original thought was to show Akka Typed in the snippets with the necessary imports for conversions where needed.

@patriknw Opinions?

Comment (patriknw) [reply build.sbt:207]: I agree, it's not worth adding the dependency just for this (rebalance listener).

Comment (seglo) [reply build.sbt:207]: Ok, I'll revert the changes to core and move the typed to classic interoperability to the docs example.

Comment (patriknw) [reply build.sbt:207]: thanks

**Review by ennru — APPROVED** [review]
LGTM, but javafmt doesn't like it.

**Comment by seglo** [comment]
javafmt can take a hike!

**Pull request merged**